File: Backend/Entities/Item.cs
namespace Backend.Entities;

public class Item
{
    public int Id { get; set; }
    public string Name { get; set; }
    public int VendorId { get; set; }
    public Vendor Vendor { get; set; }
    public int CategoryId { get; set; }
    public Category Category { get; set; }
    public int Stock { get; set; }
    public double Price { get; set; }
    public int AppUserId { get; set; } // Admin
    public AppUser AppUser { get; set; } // Admin
    public DateTime CreatedAt { get; set; }

    public ICollection<ReceiptItem> ReceiptItems { get; set; } = [];
}

File: Frontend/src/types/Item.ts
export type Item = {
  id: number;
  name: string;
  vendor: NamedApiResource;
  category: NamedApiResource;
  stock: number;
  price: number;
  createdBy: {
    id: number;
    userName: string;
    email: string;
    suspend: boolean;
  };
  createdAt: string;
}

export type NamedApiResource = {
  id: number;
  name: string;
}

